### PR TITLE
Migrate from TSLint to ESLint

### PR DESCRIPTION
**Pull request by t1m0thyj**
ℹ️ This PR is easier to review with whitespace changes hidden. It will still be a big diff though 😢 
⚠️ In cases where tests were removed, it is because they were identified as duplicates by ESLint.

TODO:
* [x] Discuss lint rules with the team
* [x] Fix or disable remaining lint errors
* [x] Fix or disable remaining lint warnings
* [x] Remove comments that disable TSLint rules
* [x] Remove TSLint from the test CLIs
* [x] Update/remove Gulp task that uses TSLint

### CODE REVIEW

**Comment by t1m0thyj** [comment]
| TSLint Rule | ESLint Equivalent | Is Migrated |
|:-:|:-:|:-:|
| no-submodule-imports | none, but [import/no-internal-modules](https://github.com/typescript-eslint/tslint-to-eslint-config/issues/1111) is similar [1] | ❌ |
| max-line-length = 150 | max-len | ✔️ |
| no-magic-numbers | yes, no-magic-numbers but I don't like it | ❌ |
| no-consecutive-blank-lines | no-multiple-empty-lines | ✔️ |
| indent = 4 spaces | indent | ✔️ |
| whitespace = check-module | none, but [object-curly-spacing](https://github.com/eslint/eslint/issues/9973) is similar [2] | ❌ |

[1] It prevents importing submodules from inside your own project, but we want to prevent importing them from other packages.
[2] It enforces spacing for curly braces everywhere, but we want to only enforce it for import/export statements.

**Comment by t1m0thyj** [comment]
After discussing with the team:
* no-submodule-imports - ok to ignore
* no-magic-numbers - add for production code, but ignore in tests
* whitespace - ok to ignore
* trailing-comma - add for multiline, but ignore for single line

File: packages/imperative/__tests__/plugins/utilities/npm-interface/install.test.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@
 *
 */
 
+/* eslint-disable jest/expect-expect */
```
Comment (zFernand0) [inline packages/imperative/__tests__/plugins/utilities/npm-interface/install.test.ts:12]: we might be able to rename the functions (like `wasNpmInstallCallValid`) which contain the expects to something like `expectNpmInstallCallValid` (or soemthing to that matches the `assertFunctionNames`) 😋

**Review by zFernand0 — APPROVED** [review]
Whitespaces certainly reduces the number of lines changed 😋 
Still takes a little while to review.
LGTM though! 😉

File: __tests__/__src__/environment/doc/parms/ISetupEnvironmentParms.ts [hunk 2]
```diff
@@ -9,14 +9,6 @@
 *
 */
 
-import { IImperativeConfig } from "../../../../../packages/imperative";
-/**
```
Comment (awharn) [inline __tests__/__src__/environment/doc/parms/ISetupEnvironmentParms.ts:13]: Was it supposed to remove this?

File: __tests__/src/TestUtil.ts [hunk 3]
```diff
@@ -164,8 +164,8 @@ export enum CMD_TYPE {
 }
 
 export function executeTestCLICommand(cliBinModule: string, testContext: any, args: string[],
```
Comment (awharn) [inline __tests__/src/TestUtil.ts:166]: Do we want to move everything to be on another line, so it's more uniform?

File: packages/cmd/__tests__/CommandProcessor.test.ts [hunk 4]
```diff
@@ -1600,46 +1608,6 @@ describe("Command Processor", () => {
         expect(commandResponse).toMatchSnapshot();
     });
 
-    it("should invoke the handler and return success=true if the handler was successful", async () => {
```
Comment (awharn) [inline packages/cmd/__tests__/CommandProcessor.test.ts:1603]: Was this test supposed to be removed?

**Review by awharn — COMMENTED** [review]
Just a couple questions before I approve this.

File: packages/cmd/__tests__/CommandProcessor.test.ts (hunk 4 above)
Comment (t1m0thyj) [reply packages/cmd/__tests__/CommandProcessor.test.ts:1603]: I think so, it appeared to be a duplicate of this test which still exists in the "use-eslint" branch: https://github.com/zowe/imperative/blob/e8b7aee9a77e2f2075423c317096feb34a011483/packages/cmd/__tests__/CommandProcessor.test.ts#L1327-L1365

**Pull request merged**